stdlib/public/SDK/Foundation/NSRange.swift

```diff
@@ -12,33 +12,117 @@
 
 @_exported import Foundation // Clang module
 
+extension NSRange : Hashable {
+    public var hashValue: Int { 
+        return Int(bitPattern: UInt(location) ^ UInt(location))
```

Comment:
yea that should have been length, I guess I could bit shift them by 32 bits and OR since NSRange (excluding NSNotFound locations) are usually very low bit values.